homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,485 @@
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error("Could not get %s state", self.name, exc_info=True)
+            self._is_available = False
+
+    async def _async_switch_on_off_executor(self, turn_on: bool) -> None:
+        """Handle profile switch."""
+        state = "unlimited" if turn_on else "never"
+        await self.hass.async_add_executor_job(
+            lambda: self.fritzbox_tools.fritz_profiles[self.profile].set_state(state)
+        )
+
+
+class FritzBoxWifiSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools Wifi switch."""
+
+    def __init__(self, fritzbox_tools, device_friendly_name, network_num, network_name):
+        """Init Fritz Wifi switch."""
+        self._fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self._attributes = {}
+        self._network_num = network_num
+
+        switch_info = SwitchInfo(
+            description=f"Wi-Fi {network_name}",
+            friendly_name=device_friendly_name,
+            icon="mdi:wifi",
+            type=SWITCH_TYPE_WIFINETWORK,
+            callback_update=self._async_fetch_update,
+            callback_switch=self._async_switch_on_off_executor,
+        )
+        super().__init__(self._fritzbox_tools, switch_info)
+
+    async def _async_fetch_update(self):
+        """Fetch updates."""
+
+        try:
+            wifi_info = await self.hass.async_add_executor_job(
+                lambda: self._fritzbox_tools.connection.call_action(
+                    f"WLANConfiguration:{self._network_num}", "GetInfo"
+                )
+            )
+            _LOGGER.debug(
+                "Specific %s response: GetInfo=%s", SWITCH_TYPE_WIFINETWORK, wifi_info
+            )
+
+            self._attr_is_on = wifi_info["NewEnable"] is True
+            self._is_available = True
+
+            self._attributes["beacon_type"] = wifi_info["NewBeaconType"]
+            self._attributes["BSSID"] = wifi_info["NewBSSID"]
+            self._attributes["mac_address_control"] = wifi_info[
+                "NewMACAddressControlEnabled"
+            ]
+
+        except FritzConnectionException:
+            _LOGGER.error(
+                "Authorization Error: Please check the provided credentials and verify that you can log into the web interface",
+                exc_info=True,
+            )
+            self._is_available = False
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error("Could not get %s state", self.name, exc_info=True)
+            self._is_available = False
+
+    async def _async_switch_on_off_executor(self, turn_on: bool) -> None:
+        """Handle wifi switch."""
+        self.hass.async_add_executor_job(
+            lambda: self._fritzbox_tools.connection.call_action(
+                f"WLANConfiguration{self._network_num}",
+                "SetEnable",
+                NewEnable="1" if turn_on else "0",
+            )
+        )
```

Comment:
could we generalize `service_call_action` to also work here?

Reply:
Indeed, as I told you on Discord we should replace all `fritzbox_tools.connection.call_action` with `service_call_action`

Reply:
Done